lib/gitlab/markdown/relative_link_filter.rb

```diff
@@ -0,0 +1,117 @@
+require 'uri'
+
+module Gitlab
+  module Markdown
+    # HTML filter that "fixes" relative links to files in a repository.
+    #
+    # Context options:
+    #   :commit
+    #   :project
+    #   :project_wiki
+    #   :requested_path
+    #   :ref
+    class RelativeLinkFilter < HTML::Pipeline::Filter
+
+      def call
+        if !project_wiki && repository.try(:exists?) && !repository.empty?
```

Comment:
Maybe extract this group of conditions into a helper method, like `relative_linkable?` to aid in testing.